src/sinks/http.rs

```diff
@@ -48,6 +48,13 @@ pub struct HttpSinkConfig {
     #[serde(default)]
     pub request: TowerRequestConfig,
     pub tls: Option<TlsOptions>,
+
+    // A parsed uri used once the original uri field has been
+    // parsed to allow access to the uri in `HttpSink::build_request`
+    // without having to have that function parse the uri each time. Users
+    // should not be able to construct this type and it has a default impl.
+    #[serde(skip)]
+    pub uri__: Option<Uri>,
```

Comment:
I'm not a big fan of this pattern. I get that it's more convenient to have a single struct, but the usage becomes less clear.

Reply:
Yeah, the other option is to create a sub struct which I can do, just thought I'd try this out.

Reply:
Removed and I added `UriSerde`.